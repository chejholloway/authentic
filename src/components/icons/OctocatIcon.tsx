import React from "react";
import octocat from "../../assets/images/github-octocat-svgrepo-com.svg";
/**
 * Renders the Octocat icon as an image.
 *
 * This component displays the Octocat logo as an image. You can provide an optional
 * CSS class name to style the icon as needed.
 *
 * @component
 * @example
 * ```tsx
 * <OctocatIcon className="text-gray-600" />
 * ```
 * @defaultprops
 * @param {object} props - The component props.
 * @param {string} [props.className] - An optional CSS class name to apply custom styling.
 */
interface OctocatIconProps {
  className?: string;
}

const OctocatIcon: React.FC<OctocatIconProps> = ({ className }) => {
  return (
    <img
      className={`h-10 object-cover ${className}`}
      src={octocat}
      alt="Github Octocat"
    />
  );
};

export default OctocatIcon;
