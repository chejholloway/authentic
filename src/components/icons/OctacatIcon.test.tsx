import React from "react";
import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer"; // Import react-test-renderer
import OctocatIcon from "./OctocatIcon";

describe("OctocatIcon", () => {
  test("renders Octocat icon", () => {
    render(<OctocatIcon className="test-class" />);

    // Assert that the Octocat icon image is rendered with the provided class
    const octocatIcon = screen.getByAltText("Github Octocat");
    expect(octocatIcon).toBeInTheDocument();
    expect(octocatIcon).toHaveClass("h-10 object-cover test-class");
  });

  test("matches snapshot", () => {
    // Use react-test-renderer to create a snapshot of the component
    const tree = renderer.create(<OctocatIcon className="test-class" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
