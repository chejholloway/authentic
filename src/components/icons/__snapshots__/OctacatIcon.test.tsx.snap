// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OctocatIcon matches snapshot 1`] = `
<img
  alt="Github Octocat"
  className="h-10 object-cover test-class"
  src="github-octocat-svgrepo-com.svg"
/>
`;
